src/modules/accommodations/accommodations.controller.ts

```diff
@@ -27,6 +49,58 @@ export class AccommodationsController {
     };
   }
 
+  @Post()
+  @CreateAccommodationSwaggerDecorator()
+  @UseGuards(AuthTokenGuard)
+  async createAccommodation(
+    @Body() createDto: CreateAccommodationDto,
+    @GetUser() payload: JwtPayload,
+  ) {
+    const newAccommodation = this.accommodationsService.createAccommodation(
+      createDto,
+      payload.sub,
+    );
+    return { data: newAccommodation };
+  }
+
+  @Get(':id')
+  @GetByIdSwaggerDecorator()
+  async getAccommodationById(@Param('id', ParseUUIDV4Pipe) id: string) {
+    const accommodationData =
+      this.accommodationsService.getAccommodationById(id);
+    return { data: accommodationData };
+  }
+
+  @Put(':id')
+  @UpdateAccommodationSwaggerDecorator()
+  @UseGuards(AuthTokenGuard)
+  async updateAccommodation(
+    @Param('id', ParseUUIDV4Pipe) id: string,
+    @Body() updateDto: UpdateAccommodationDto,
+    @GetUser() payload: JwtPayload,
```

Comment:
Same. Check what GetUser is returning and change the type accordingly, or leave if no mistake here.